src/Provider/AbstractProvider.php

```diff
@@ -217,12 +227,8 @@ public function getAccessToken($grant = 'authorization_code', $params = [])
 
                 break;
         }
-
-        if (isset($result['error']) && ! empty($result['error'])) {
-            // @codeCoverageIgnoreStart
-            throw new IDPException($result);
-            // @codeCoverageIgnoreEnd
-        }
+        $checker = $this->errorChecker;
+        $checker($result);
```

Comment:
Yes, errorChecker must be callable. the setter of errorChecker has type-hint of Closure. After rethinking, i think my code is terrible. thephpleague/oauth2-client  v1 is oriented to third party provider authors and should easy for these authors to extend. Maybe add a abstract method errorChecker in  AbstractProvider is better and enough. So third party provider authors can define what is error according to third party provider document.